btp.lua

```diff
@@ -0,0 +1,486 @@
+local btp_proto = Proto("btp", "Bitnomial Trading Protocol")
+
+local disconnect_reason_string = {
+    [0x01] = "Sequence ID fault",
+    [0x02] = "Heartbeat fault",
+    [0x03] = "Failed to login within one (1) second",
+    [0x04] = "Messaging rate exceeded",
+    [0x05] = "Failed to parse message"
+}
+
+local login_reject_reason_string = {
+    [0x01] = "Not logged in",
+    [0x02] = "Unauthorized",
+    [0x03] = "Already logged in"
+}
+
+local order_entry_reject_reason_string = {
+    [0x01] = "Account not found",
+    [0x02] = "Product not found",
+    [0x03] = "Order not found",
+    [0x04] = "Order already exists",
+    [0x05] = "Order already closed",
+    [0x06] = "Order not changed by modify",
+    [0x07] = "Quantity greater than max order size",
+    [0x08] = "Quantity less than min order size",
+    [0x09] = "Price outside price bands",
+    [0x0A] = "Price outside price limits",
+    [0x0B] = "Price not tick aligned (reserved)",
+    [0x0C] = "Market halted. Only close requests accepted",
+    [0x0D] = "Market closed. No requests accepted",
+    [0x0E] = "Give-up account not found",
+    [0x0F] = "Give-up unauthorized",
+    [0x10] = "Messaging rate exceeded",
+    [0x11] = "Position limit exceeded",
+    [0x12] = "Connection disabled"
+}
+
+local market_state_string = {
+    [string.byte("O")] = "Open",
+    [string.byte("H")] = "Halt",
+    [string.byte("C")] = "Closed"
+}
+
+local firm_type_string = {
+    [string.byte("R")] = "Reg/House",
+    [string.byte("S")] = "Seg/Customer"
+}
+
+local liquidity_string = {
+    [string.byte("A")] = "Add",
+    [string.byte("R")] = "Remove",
+    [string.byte("S")] = "Spread leg match"
+}
+
+local side_string = {[string.byte("A")] = "Ask", [string.byte("B")] = "Bid"}
+
+local close_reason_string = {
+    [string.byte("I")] = "IOC finished",
+    [string.byte("G")] = "Non-connection cancel",
+    [string.byte("S")] = "Self-match prevention cancel"
+}
+
+local time_in_force_string = {
+    [string.byte("D")] = "Day",
+    [string.byte("I")] = "IOC"
+}
+
+local order_entry_message_type_string = {
+    [string.byte("O")] = "Open",
+    [string.byte("M")] = "Modify",
+    [string.byte("A")] = "Ack",
+    [string.byte("R")] = "Reject",
+    [string.byte("C")] = "Closed",
+    [string.byte("F")] = "Fill"
+}
+
+local drop_copy_message_type_string = {
+    [string.byte("O")] = "Open Ack",
+    [string.byte("M")] = "Modify Ack",
+    [string.byte("F")] = "Fill",
+    [string.byte("C")] = "Close"
+}
+
+local price_feed_message_type_string = {
```

Comment:
nit: We've typically had `pricefeed` as one word in our system. Maybe a little strange, but that's what we do.